src/App.tsx

```diff
@@ -14,11 +14,110 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+interface State {
+  originGoods: string[],
+  goods: string[],
+  isVisible: boolean,
+  isReverse: boolean,
+  isSortByAlphabet: boolean,
+  isSortByLength: boolean,
+}
+
+export default class App extends React.Component<{}, State> {
+  state:State = {
+    originGoods: [...goodsFromServer],
```

Comment:
no need to save goodsFromServer inside state, just use goodsFromServer inside the component